src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private transient Node<K, V>[] table;
+    private int size;
+    private double threshhold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        int size;
+        threshhold = table.length * LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > threshhold) {
+            resize();
+        }
+        Node newNode = new Node<>(key, value, null);
+        int index = index(key);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> node = table[index];
+            if (Objects.equals(node.key,key)) {
+                node.value = value;
+                return;
+            }
+            while (node.next != null) {
+                if (Objects.equals(node.next.key, key)) {
+                    node.next.value = value;
+                    return;
+                }
+                node = node.next;
+            }
+            node.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = index(key);
+        if (table[index] == null) {
+            return null;
+        }
+        Node<K,V> node = table[index];
+        if (Objects.equals(node.key,key)) {
+            return node.value;
+        }
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int index(K key) {
```

Comment:
```suggestion
    private int getIndex(K key) {
```
use verbs for method names